File: articles/punishment-forgiveness-and-moral-repair/cpp/moral_repair_score.cpp
/*
Synthetic moral repair score example.
Educational scaffold only.

Compile:
  c++ -std=c++17 cpp/moral_repair_score.cpp -o outputs/tables/moral_repair_score_cpp
Run:
  ./outputs/tables/moral_repair_score_cpp
*/

#include <iostream>

int main() {
    double punishment = 0.4;
    double forgiveness_probability = 0.6;
    double restitution = 0.8;
    double acknowledgment = 0.9;
    double victim_recognition = 0.7;
    double institutional_trust = 0.5;
    double structural_reform = 0.6;
    double ongoing_threat = 0.3;

    double repair =
        0.20 * punishment +
        0.25 * forgiveness_probability +
        0.30 * restitution +
        0.30 * acknowledgment +
        0.25 * victim_recognition +
        0.20 * institutional_trust +
        0.25 * structural_reform -
        0.25 * ongoing_threat;

    std::cout << "Synthetic moral repair score: " << repair << std::endl;
    return 0;
}

File: articles/punishment-forgiveness-and-moral-repair/go/moral_repair_score.go
// Synthetic moral repair score example.
// Educational scaffold only.
//
// Run:
//   go run go/moral_repair_score.go

package main

import "fmt"

func main() {
	punishment := 0.4
	forgivenessProbability := 0.6
	restitution := 0.8
	acknowledgment := 0.9
	victimRecognition := 0.7
	institutionalTrust := 0.5
	structuralReform := 0.6
	ongoingThreat := 0.3

	repair := 0.20*punishment +
		0.25*forgivenessProbability +
		0.30*restitution +
		0.30*acknowledgment +
		0.25*victimRecognition +
		0.20*institutionalTrust +
		0.25*structuralReform -
		0.25*ongoingThreat

	fmt.Printf("Synthetic moral repair score: %.3f\n", repair)
}
